### PR TITLE
Improve Error Handler for Results Uploader

### PR DESCRIPTION
**Pull request by chaseconey**
This PR:

- Adds additional checks for the upload process
- Adds tests for all of the edge cases
- Makes local testing not wipe out local data

### CODE REVIEW

File: app/Traits/RaceResultsParser.php [hunk 1]
```diff
@@ -8,6 +8,7 @@
 use App\Models\F1Team;
 use App\Models\Race;
 use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
```
Comment (chaseconey) [inline app/Traits/RaceResultsParser.php:11]:
```suggestion

```
